code/processes/gateway.q

```diff
@@ -421,7 +421,7 @@ asyncexec:asyncexecjpt[;;raze;();0Wn]
 // execute a synchronous query
 syncexecjpre36:{[query;servertype;joinfunction]
  // Check correct function called
- if[(.z.w<>0)&(.z.w in key .gw.call)&not .gw.call .z.w;
+ if[(.z.w<>0)&(.z.w in key .gw.call)&(not .gw.call .z.w)& not @[value;`.kxdash.enabled;0b];
```

Comment:
this will allow `syncexec` to be called via async request as long as `.kxdash.enabled` is true i.e. it'd allow a non-kx dashboard client to send async request to `syncexec`, which I don't think is what we want?

it seems strange to me that we are using `syncexecjpre36` for async requests from kx dashboards, is this definitely the correct approach?

Reply:
@davidbuckley123 couple of points to pick up here if you get a chance

Reply:
> non-kx dashboard client to send async request to syncexec
Fair enough - the `if[]` statement could be tightened up a bit.

I am only trying to enabled syncexec calls from a negative handle from kx dashboards.
Honestly, I am not 100% satisfied with this as a solution.
I'll take some time and see if I can come up with something more elegant.